gk/main.c

```diff
@@ -1139,14 +1146,15 @@ xmit_icmpv6(struct rte_mbuf *pkt, uint16_t *num_pkts,
  */
 static int
 update_ip_hop_count(struct ipacket *packet, uint16_t *num_pkts,
-	struct rte_mbuf **icmp_bufs)
+	struct rte_mbuf **icmp_bufs, struct token_bucket_ratelimit_state *rs)
 {
 	if (packet->flow.proto == ETHER_TYPE_IPv4) {
 		struct ipv4_hdr *ipv4_hdr =
 			rte_pktmbuf_mtod_offset(packet->pkt,
 			struct ipv4_hdr *, sizeof(struct ether_hdr));
 		if (ipv4_hdr->time_to_live <= 1) {
-			xmit_icmp(packet->pkt, num_pkts, icmp_bufs);
+			if (!tb_ratelimit(rs))
```

Comment:
Both callers of `tb_ratelimit()` invert the return and the kernel function even have the return inverted as well, so why don't we invert the return of `tb_ratelimit()` and simplify the callers?